cluster/grpc_rpc_client.go

```diff
@@ -201,14 +205,18 @@ func (gs *GRPCClient) SendPush(userID string, frontendSv *Server, push *protos.P
 func (gs *GRPCClient) AddServer(sv *Server) {
 	var host, port string
 	var ok bool
-	if host, ok = sv.Metadata["grpc-host"]; !ok {
-		logger.Log.Errorf("server %s doesn't have a grpc-host specified in metadata", sv.ID)
+
+	host = gs.getServerHost(sv)
+	if host == "" {
+		logger.Log.Errorf("server %s has no grpc-host specified in metadata", sv.ID)
 		return
 	}
-	if port, ok = sv.Metadata["grpc-port"]; !ok {
-		logger.Log.Errorf("server %s doesn't have a grpc-port specified in metadata", sv.ID)
+
+	if port, ok = sv.Metadata[constants.GRPCPortKey]; !ok {
```

Comment:
Does the internal port need to be the same as the external one?

Reply:
Two different ports wouldn't add an unnecessary complexity to the configuration?
But maybe it's better to be more flexible and accept different ports

Reply:
I feel it's strange to force the same port for internal and external access, but if the additional complexity is too high we can leave it like this for now